hybrid/hybrid_simulation.py

```diff
@@ -265,16 +265,26 @@ def calculate_financials(self):
             return
 
         size_ratios = []
-
         for v in generators:
             size_ratios.append(v.system_capacity_kw / hybrid_size_kw)
-        assert abs(sum(size_ratios) - 1) < 1e-7
 
-        def set_average_for_hybrid(var_name):
+        production_ratios = []
+        total_production = sum([v.annual_energy_kw for v in generators])
+        for v in generators:
+            production_ratios.append(v.annual_energy_kw / total_production)
```

Comment:
How does the production_ratio work when the battery returns a negative value?

Reply:
I set up a `non_storage_production_ratio` for the O&M variable costs, which are computed differently for storage and non-storage generation. For PV and wind, the `om_production` is applied to the `gen` vector, and for battery, the `om_batt_variable_cost` is applied to just the battery discharge.

I added new tests to make sure these variable costs and the production tax credits make sense:
https://github.com/NREL/HOPP/blob/0d990492c12497f1a4b39b0ba8a5403be2805901/tests/hybrid/test_hybrid.py#L187